docs/en/docs/encoders.md

```diff
@@ -0,0 +1,202 @@
+# Encoders
+
+Encoders are basically converters between datastructures and json. Its main use is in response were different return values are automatically
+parsed into json objects.
+
+For understanding what they are doing, it is helpful to see the manual usage
+
+## Manual usage
+
+Encoders provide two public methods:
+
+- json_encode
+- apply_structure
+
+### json_encode
+
+`json_encode` is basically an enhanced json.dumps. It provides via the encoders converters which parse additional
+types to json.
+By default only a simplification is done, therefore the json string is deserialized again so a simple datastructure is returned.
+
+``` python
+{!> ../../../docs_src/encoders/direct_usage.py !}
+```
+
+### apply_structure
+
+`apply_structure` is basically the inverse of `json_encode`.
+It assumes it is known from which structure a value was serlialized.
+If one encoder matches via `is_type_structure` (for Encoder by default `isclass(structure) and issubclass(structure, self.__type__)`) the encoder is used.
+First is checked via `is_type` if the value is already converted and if yes simply returned.
+Otherwise the value is molded via the structure in an instance and returned.
+
+``` python
+{!> ../../../docs_src/encoders/molding.py !}
+```
+
+## Default Encoders
+
+In order to understand how to serialize a specific object into `json`, Lilya has some default
+encoders that evaluates when tries to *guess* the response type.
+
+* `DataclassEncoder` - Serialises `dataclass` objects.
+* `NamedTupleEncoder` - Serialises `NamedTuple` objects.
+* `ModelDumpEncoder` - Serialises objects by calling its model_dump method. This allows serializing pydantic objects out of the box.
+* `EnumEncoder` - Serialises `Enum` objects.
+* `PurePathEncoder` - Serializes `PurePath` objects.
+* `DateEncoder` - Serializes date and datetime objects.
+* `StructureEncoder` - Serializes more complex data types which implement `Iterable`.
+
+What a brand new encoder is needed and it is not natively supported by Lilya? Well, [building a custom encoder](#build-a-custom-encoder)
```

Comment:
What if a brand new encoder is needed and it is not natively supported by Lilya?